documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -1334,6 +1334,26 @@ include::{testDir}/example/ParameterizedTestDemo.java[tags=ArgumentsSource_examp
 include::{testDir}/example/ParameterizedTestDemo.java[tags=ArgumentsProvider_example]
 ----
 
+[[writing-tests-parameterized-tests-sources-multi-valued-arguments-source]]
+A bit more interesting example shows ArgumentsSource can do a similar parameter interpretation as
+CsvSource. As such it is even more powerful than CsvSource, because it can pass values that are not easily
+converted from String into the Java code domain, such as  *Lambda expressions*.
+
+[source,java,indent=0]
+----
+include::{testDir}/example/ParameterizedTestDemo.java[tags=ArgumentsSource2_example]
+----
+
+NOTE: You have to pay attention to the fact that the order of the values in the arrays below matches the order of
+the parameters in the method, in this example `testWithArgumentsSource2` above.
+
+[source,java,indent=0]
+----
+include::{testDir}/example/ParameterizedTestDemo.java[tags=LambdaArgumentsProvider_example]
+----
+
+Note that the example shows the Lambda expressions to be assigned to values of the appropriate type, like `IntBinaryOperation plus`. This
```

Comment:
If the team decides to accept this PR, the documentation will need to be wrapped at 90 characters according to [project guidelines](https://github.com/junit-team/junit5/blob/main/CONTRIBUTING.md#documentation).